### PR TITLE
Recursively import all has_one/has_many associations

### PR DESCRIPTION
**Pull request by sferik**
This should work for all `has_one`/`has_many` associations, not just ones explicitly marked `autosave`.

### CODE REVIEW

File: lib/activerecord-import/import.rb [hunk 1]
```diff
@@ -462,7 +461,7 @@ def import_associations(models, options)
     def find_associated_objects_for_import(associated_objects_by_class, model)
       associated_objects_by_class[model.class.name]||={}
 
-      model.class.reflect_on_all_autosave_associations.each do |association_reflection|
+      model.class.reflect_on_all_associations(:has_many).each do |association_reflection|
```
Comment (jkowens) [inline lib/activerecord-import/import.rb]: I think this might need to be revised to include :has_one associations.

**Comment by jkowens** [comment]
:+1:

File: lib/activerecord-import/import.rb (hunk 1 above)
Comment (sferik) [inline lib/activerecord-import/import.rb]: Yes. That can be added once #234 is merged.

Comment (zdennis) [inline lib/activerecord-import/import.rb]: @sferik , #234 just got merged in. :)

**Comment by sferik** [comment]
@zdennis I just rebased this branch from `master` and added back support for `has_one` associations.

**Comment by zdennis** [comment]
Awesome! Thanks @sferik!

**Comment by sferik** [comment]
As you’re well aware, the tests are failing due to an [issue with `em-synchrony`](https://github.com/igrigorik/em-synchrony/issues/201), which you reported one month ago and has received no response. That project shows no sign of being updated to work with Active Record 5. According to [this open issue](https://github.com/igrigorik/em-synchrony/issues/198), it still doesn’t work with Active Record 4.2.1, which was released on March 19, 2015. In fact, the last commit to `em-synchrony` was on March 16, 2015, almost a year ago.

At this point, I think dropping support for `em-synchrony` is the right thing to do for this project. If someone wants to use `activerecord-import` with `em-synchrony`, they can take ownership over getting the test build passing with it. In the mean time, I will open a pull request that removes `em-synchrony` from the test suite and turns the build green. You can merge it if/when you agree with this assessment.

**Comment by zdennis** [comment]
Closing this in favor of #243 which builds on this PR. Thanks for your work on this @sferik.

Also `em-synchrony` and `mysql` have been removed in #239.

**Comment by zdennis** [comment]
This has been included in the [0.12.0](https://rubygems.org/gems/activerecord-import/versions/0.12.0) release.

**Pull request closed without merging**